code/go/0chain.net/smartcontract/minersc/update_test.go

```diff
@@ -0,0 +1,289 @@
+package minersc
+
+import (
+	"testing"
+
+	configpkg "0chain.net/chaincore/config"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func assertErrMsg(t *testing.T, err error, msg string) {
+	t.Helper()
+
+	if msg == "" {
+		assert.Nil(t, err)
+		return
+	}
+
+	if assert.NotNil(t, err) {
+		assert.Equal(t, msg, err.Error())
+	}
+}
+
+func getGlobalNodeTest() (gn *GlobalNode) {
+	const pfx = "smart_contracts.minersc."
+
+	configpkg.SmartContractConfig.Set(pfx+"max_stake", 1000)
+	configpkg.SmartContractConfig.Set(pfx+"min_stake", 1)
+	configpkg.SmartContractConfig.Set(pfx+"max_n", 50)
+	configpkg.SmartContractConfig.Set(pfx+"min_n", 2)
+	configpkg.SmartContractConfig.Set(pfx+"t_percent", .51)
+	configpkg.SmartContractConfig.Set(pfx+"k_percent", .75)
+	configpkg.SmartContractConfig.Set(pfx+"max_s", 30)
+	configpkg.SmartContractConfig.Set(pfx+"min_s", 1)
+	configpkg.SmartContractConfig.Set(pfx+"max_delegates", 20)
+	configpkg.SmartContractConfig.Set(pfx+"reward_round_frequency", 250)
+	configpkg.SmartContractConfig.Set(pfx+"interest_rate", .1)
+	configpkg.SmartContractConfig.Set(pfx+"reward_rate", .11)
+	configpkg.SmartContractConfig.Set(pfx+"share_ratio", .27)
+	configpkg.SmartContractConfig.Set(pfx+"block_reward", 3000)
+	configpkg.SmartContractConfig.Set(pfx+"max_charge", .44)
+	configpkg.SmartContractConfig.Set(pfx+"epoch", 10000)
+	configpkg.SmartContractConfig.Set(pfx+"reward_decline_rate", .33)
+	configpkg.SmartContractConfig.Set(pfx+"interest_decline_rate", .34)
+	configpkg.SmartContractConfig.Set(pfx+"max_mint", 3)
+
+	return &GlobalNode{
+		0, 50, 2, 30, 1, 20, .51, .75, 0, 1000 * 1e10, 1e10,
+		.1, .11, .27, 3000 * 1e10, .44, 10000, .33, .34, 3 * 1e10,
+		nil, 0, 250,
+	}
+}
+
+func Test_getConfig(t *testing.T) {
+	var (
+		msc        = newTestMinerSC()
+		balances   = newTestBalances()
+		configured = getGlobalNodeTest()
+		gn, _      = msc.getGlobalNode(balances)
+	)
+	assert.EqualValues(t, configured, gn)
+}
+
+func TestMinerSmartContractUpdate(t *testing.T) {
+	var (
+		msc        = newTestMinerSC()
+		balances   = newTestBalances()
+		tx         = newTransaction(owner, msc.ID, 0, 0)
+		originalGn = getGlobalNodeTest()
+		gn, err    = msc.getGlobalNode(balances)
+		update     = &GlobalNode{}
+	)
+
+	balances.txn = tx
+
+	// 1. Malformed update
+	t.Run("malformed update", func(t *testing.T) {
```

Comment:
For consistancy can now we adopt the [table driven subtest style](https://blog.golang.org/subtests#TOC_4.) of @MurashovVen's unit tests.